tracer/src/Datadog.InstrumentedAssemblyVerification/IlSpyDecompilationVerification.cs

```diff
@@ -42,36 +43,25 @@ public List<string> Verify()
             var errors = new List<string>();
             var settings = new DecompilerSettings();
             var decompiler = new CSharpDecompiler(_peFile, settings);
-            foreach (string method in _methods)
+            foreach ((string type, string method) in _methods)
             {
                 try
                 {
-                    int methodNameIndex = method.Substring(0, method.LastIndexOf('(')).LastIndexOf(".", StringComparison.InvariantCulture);
-                    string methodAndParametersName = method.Substring(methodNameIndex + 1);
-                    FullTypeName typeName;
-                    if (methodAndParametersName.StartsWith("ctor") || methodAndParametersName.StartsWith("cctor"))
-                    {
-                        typeName = new FullTypeName(method.Substring(0, methodNameIndex - 1));
-                    }
-                    else
-                    {
-                        typeName = new FullTypeName(method.Substring(0, methodNameIndex));
-                    }
-                    ITypeDefinition typeInfo = decompiler.TypeSystem.MainModule.Compilation.FindType(typeName).GetDefinition();
+                    ITypeDefinition typeInfo = decompiler.TypeSystem.MainModule.Compilation.FindType(new FullTypeName(type)).GetDefinition();
                     if (typeInfo == null)
                     {
-                        string error = $"{nameof(IlSpyDecompilationVerification)} Type {typeName} not found in assembly {decompiler.TypeSystem.MainModule.AssemblyName}";
+                        string error = $"{nameof(IlSpyDecompilationVerification)} Type {type} not found in assembly {decompiler.TypeSystem.MainModule.AssemblyName}";
                         _logger.Warn(error);
                         continue;
                     }
 
                     // we can't do this replace in ILSpy method because '`' is valid sign for generic type
-                    var copyOfMethodAndParameterName = methodAndParametersName.Replace("!", "`");
```

Comment:
This seems to be duplicating code that already exists in `ModifiedMethod.Decompile`, perhaps extract it to a method under `ILSpyHelper`?

Reply:
Fixed in d6e34e581c70cc2271806c12db2c3d4f2caf78f1